spec/requests/posts_spec.rb

```diff
@@ -1,7 +1,44 @@
 require 'rails_helper'
 
-RSpec.describe 'Posts', type: :request do
-  describe 'GET /index' do
-    pending "add some examples (or delete) #{__FILE__}"
+RSpec.describe PostsController, type: :controller do
+  before { get :index, params: { user_id: 1 } }
+  it { should respond_with(200) }
+end
+
+RSpec.describe PostsController, type: :controller do
+  it { should route(:get, 'users/1/posts').to(action: :index, user_id: 1) }
```

Comment:
- [ ] Great job 👏🏻 However, you should use `type: request` as shown in the example code OR use `config.infer_spec_type_from_file_location!` in the Rspec config file (in that case just placing a file in the `spec/requests` dir will be enough).  `config.infer_spec_type_from_file_location!` is a configuration setting that allows RSpec to automatically infer the type of a spec file based on its location in the directory structure.
- So, to ensure that a spec file is treated as a request spec, you can either explicitly specify the type of the spec using type: :request in the spec file itself, like so:

   ```ruby
    require 'rails_helper'
   
    RSpec.describe "Users", type: :request do
      ...
    end
   ```
- Make sure. `index` and `show` actions for `User` and `Post` controllers are tested

   ```ruby
   describe 'GET /index' do
     ...
   end
   
   describe 'GET /show' do
     ...
   end
   ```

- You could make reference to this [PR](https://github.com/Nkaleth/blog_web/pull/6) if you have any question.

Reply:
Sorry, I missed that point. Thanks for pointing out. Corrected accordingly.